src/Features/CSharp/Portable/AddAccessibilityModifiers/CSharpAddAccessibilityModifiersDiagnosticAnalyzer.cs

```diff
@@ -78,16 +78,54 @@ public CSharpAddAccessibilityModifiersDiagnosticAnalyzer()
 
             // If they already have accessibility, no need to report anything.
             var accessibility = generator.GetAccessibility(member);
-            if (accessibility != Accessibility.NotApplicable)
+
+            var isOmit = option.Value == AccessibilityModifiersRequired.OmitIfDefault;
+
+            if (isOmit)
             {
-                return;
+                if (accessibility == Accessibility.NotApplicable)
+                {
+                    return;
+                }
+
+                // Check for default modifiers in namespace and outside of namespace
+                var parentKind = member.Parent.Kind();
```

Comment:
:bulb: Consider doing a switch on parent kind to harden this code, and always returning if the parent isn't a known kind. This will e.g. avoid having false diagnostics reported if default interface members get implemented.